packages/request-client.js/src/http-metamask-data-access.ts

```diff
@@ -52,7 +52,7 @@ export default class HttpMetaMaskDataAccess extends HttpDataAccess {
 
     // Creates a local or default provider
     this.provider = web3
-      ? new ethers.providers.Web3Provider(web3.currentProvider)
+      ? new ethers.providers.Web3Provider(web3)
```

Comment:
`currentProvider` is not used anymore, see https://docs.metamask.io/guide/provider-migration.html#window-ethereum-api-changes